word_racer/session.py

```diff
@@ -0,0 +1,295 @@
+import asyncio
+from collections import Counter
+from redbot.core.utils.chat_formatting import box
+from redbot.core.data_manager import cog_data_path
+import discord
+import numpy as np
+import pathlib
+import re
+from PIL import Image, ImageDraw, ImageFont
+
+__all__ = ["WordRacerSession"]
+
+_LEVEL_COUNT = 4
+
+_LEVELS = ["#######....##....##....##....#######",
+           "##..###....#............#....###..##",
+           "....##....##............##....##....",
+           "..............##....##.............."]
+
+_FREQ = [[142,25,66,53,218,15,42,34,168,1,8,88,44,130,124,45,1,135,180,125,50,10,7,2,23,4],
+         [130,27,65,53,199,17,42,35,154,2,10,86,44,118,114,45,2,123,164,115,50,12,9,3,24,6],
+         [119,29,64,53,181,19,43,37,142,3,13,84,45,108,104,46,3,112,150,105,51,15,12,5,26,8],
+         [100,33,62,53,150,24,44,40,120,9,21,80,46,90,87,47,9,93,125,88,52,23,20,12,30,16]]
+
+_BONUSES = [{}, {(0,2):2,(5,3):2}, {(0,0):3,(5,5):3}, {(0,0):3,(0,5):2,(5,0):2,(5,5):3}]
+
+class WordRacerSession:
+    def __init__(self, ctx):
+        self.level = 0
+        self.dataDir = pathlib.Path(__file__).parent.resolve() / 'data'
+        
+        # feel free to experiment with this
+        self.dictDir = self.dataDir/"dict/enable2k.txt"
+        self.fontDir = self.dataDir/"fonts/Roboto-Medium.ttf"
+
+        self.ctx = ctx
+        self.output_image_path = self.dataDir / f'board-{ctx.channel.id}.png'
+        self.scores = Counter()
+        self.round_scores = Counter()
+        self.claims = {}
+        self.valid_words = Counter()
+        self.board = [["_" for _ in range(6)] for __ in range(6)]
+        self.bonus = {}
+        self.nrows = 6
+        self.ncols = 6
+        self.round_finish = False
+        self.reaction_queue = []
```

Comment:
I'm not sure whether it'll be relevant yet, but there is an `asyncio.Queue` which may be more appropriate here.

Reply:
Ah, didn't know about that. I was modeling the queuing system from PlaySet